src/cache/cache_las.c

```diff
@@ -793,13 +802,36 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
 
             cursor->set_value(cursor, upd->durable_ts, upd->prepare_state, upd->type, &las_value);
 
+            /*
+             * Set the current update start time pair as the commit time pair of the LAS
+             * transaction.
+             */
+            if (upd->start_ts != WT_TS_NONE)
+                WT_ERR(__wt_txn_set_commit_timestamp(session, upd->start_ts));
+            cache->las_txnid = upd->txnid;
+
             /*
              * Using update instead of insert so the page stays pinned and can be searched before
              * the tree.
              */
             WT_ERR(cursor->update(cursor));
             ++insert_cnt;
 
+            /* Append a delete record to represent stop time pair for the above insert record */
```

Comment:
Hi Hari, the change is related to the writing newest committed version to disk I'm working on and I'm confused by your change. I don't think this is correct. The commit timestamp will become smaller and smaller because we iterate from the newest update to the oldest update and I think __wt_txn_set_commit_timestamp only allows commit timestamp to be set if it's larger or equal to the first_commit_timestamp by looking at the code. Also Though I read the PR description, I still don't understand why we have to add a delete pair. If we only write committed updates to the history store, we should have enough information to determine the stop time pair when we insert it. Probably you can explain this to me tomorrow.

Reply:
Hi Hari, I now understand what you are doing after the discussion with Vamsi.